examples/Azure_IoT_Central_ESP32/Azure_IoT_Central_ESP32.ino

```diff
@@ -0,0 +1,600 @@
+  mqtt_config.buffer_size = 1024;
+  mqtt_config.cert_pem = (const char*)ca_pem;
+
+  LogInfo("MQTT client target uri set to '%s'", mqtt_broker_uri);
+
+  mqtt_client = esp_mqtt_client_init(&mqtt_config);
+
+  if (mqtt_client == NULL)
+  {
+    LogError("esp_mqtt_client_init failed.");
+    result = 1;
+  }
+  else
+  {
+    esp_err_t start_result = esp_mqtt_client_start(mqtt_client);
+  
+    if (start_result != ESP_OK)
+    {
+      LogError("esp_mqtt_client_start failed (error code: 0x%08x).", start_result);
+      result = 1;
+    }
+    else
+    {
+      *mqtt_client_handle = mqtt_client;
+      result = 0;
+    }
+  }
+
+  return result;
+}
+
+/*
+ * See the documentation of `mqtt_client_deinit_function_t` in AzureIoT.h for details.
+ */
+static int mqtt_client_deinit_function(mqtt_client_handle_t mqtt_client_handle)
+{
+  int result = 0;
+  esp_mqtt_client_handle_t esp_mqtt_client_handle = (esp_mqtt_client_handle_t)mqtt_client_handle;
+
+  LogInfo("MQTT client being disconnected.");
+  
+  if (esp_mqtt_client_stop(esp_mqtt_client_handle) != ESP_OK)
+  {
+    LogError("Failed stopping MQTT client.");
+  }
+
+  if (esp_mqtt_client_destroy(esp_mqtt_client_handle) != ESP_OK)
+  {
+    LogError("Failed destroying MQTT client.");
+  }
+
+  if (azure_iot_mqtt_client_disconnected(&azure_iot) != 0)
+  {
+    LogError("Failed updating azure iot client of MQTT disconnection.");      
+  }
+
+  return 0;
+}
+
+/*
+ * See the documentation of `mqtt_client_subscribe_function_t` in AzureIoT.h for details.
+ */
+static int mqtt_client_subscribe_function(mqtt_client_handle_t mqtt_client_handle, const uint8_t* topic, size_t topic_length, mqtt_qos_t qos)
+{
+  LogInfo("MQTT client subscribing to '%s'", topic);
+       
+  // As per documentation, `topic` always ends with a null-terminator.
+  // esp_mqtt_client_subscribe returns the packet id or negative on error already, so no conversion is needed.
+  int packet_id = esp_mqtt_client_subscribe((esp_mqtt_client_handle_t)mqtt_client_handle, (const char*)topic, (int)qos);
+
+  return packet_id;
+}
+
+/*
+ * See the documentation of `mqtt_client_publish_function_t` in AzureIoT.h for details.
+ */
+static int mqtt_client_publish_function(mqtt_client_handle_t mqtt_client_handle, mqtt_message_t* mqtt_message)
+{
+  LogInfo("MQTT client publishing to '%s'", az_span_ptr(mqtt_message->topic));
+
+  int mqtt_result = esp_mqtt_client_publish(
+    (esp_mqtt_client_handle_t)mqtt_client_handle, 
+    (const char*)az_span_ptr(mqtt_message->topic), // topic is always null-terminated.
+    (const char*)az_span_ptr(mqtt_message->payload), 
+    az_span_size(mqtt_message->payload),
+    (int)mqtt_message->qos, 
+    MQTT_DO_NOT_RETAIN_MSG);
+  
+  if (mqtt_result == -1)
+  {
+    return RESULT_ERROR;
+  }
+  else
+  {
+    return RESULT_OK;
+  }
+}
+
+/* --- Other Interface functions required by Azure IoT --- */
+
+/*
+ * See the documentation of `hmac_sha512_encryption_function_t` in AzureIoT.h for details.
```

Comment:
yea this says sha512 but the function says sha256

Reply:
yea it should say 256

Reply:
Fixed everywhere.